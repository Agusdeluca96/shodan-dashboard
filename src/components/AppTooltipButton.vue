<template>
  <v-tooltip bottom>
    <template v-slot:activator="{ on, attrs }">
      <v-btn
        :loading="loading"
        :color="color ? color : 'primary'"
        @click="handleClick"
        v-bind="attrs"
        v-on="on"
        :text="!contained"
      >
        <p v-if="text" class="text-subtitle-2">{{ text }}</p>
        <v-icon v-if="icon">{{ icon }}</v-icon>
      </v-btn>
    </template>
    <span>{{ tooltip }}</span>
  </v-tooltip>
</template>

<script>
export default {
  props: {
    text: {
      type: String,
      default: ""
    },
    icon: {
      type: String,
      default: ""
    },
    tooltip: {
      type: String,
      default: ""
    },
    onClickEmit: {
      type: String,
      required: true
    },
    loading: {
      type: Boolean,
      default: false
    },
    contained: {
      type: Boolean,
      default: true
    },
    color: {
      type: String,
      default: null
    }
  },

  methods: {
    handleClick() {
      this.$emit(this.onClickEmit);
    }
  }
};
</script>
