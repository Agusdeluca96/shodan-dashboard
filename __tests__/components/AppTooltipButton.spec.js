import Vuetify from "vuetify";
import { createLocalVue, mount } from "@vue/test-utils";
import AppTooltipButton from "@/components/AppTooltipButton";

describe("AppTooltipButton.vue", () => {
  const localVue = createLocalVue();
  let vuetify;

  const wrapperFactory = (wrapperProps = { onClickEmit: "TestEvent" }) => {
    return mount(AppTooltipButton, {
      localVue,
      vuetify,
      propsData: wrapperProps
    });
  };

  beforeEach(() => {
    vuetify = new Vuetify();
  });

  it("should match snapshot", () => {
    const wrapper = wrapperFactory();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should emit specific event when the tooltip v-btn is clicked", async () => {
    const onClickEmitEvent = "TestEvent";
    const wrapper = wrapperFactory({ onClickEmit: onClickEmitEvent });
    const button = wrapper.find(".v-btn");

    // Assert no the event has not been emitted yet
    expect(wrapper.emitted()[onClickEmitEvent]).toBeFalsy();

    // Simulate a click on the button
    button.trigger("click");

    // Assert event has been emitted
    expect(wrapper.emitted()[onClickEmitEvent]).toBeTruthy();
  });
});
