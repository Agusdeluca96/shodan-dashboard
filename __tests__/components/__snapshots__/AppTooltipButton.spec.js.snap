// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppTooltipButton.vue should match snapshot 1`] = `<span class="v-tooltip v-tooltip--bottom"><!----><button type="button" class="v-btn v-btn--contained theme--light v-size--default primary" role="button" aria-haspopup="true" aria-expanded="false"><span class="v-btn__content"></span></button></span>`;
